File: services/Portfolio/Skarbiec.Portfolio/Features/Transfers/TransferDirection.cs
namespace Skarbiec.Portfolio.Features.Transfers;

/// <summary>Which side of a transfer a leg is, seen from its own asset.</summary>
public enum TransferDirection
{
    /// <summary>The source's Withdraw leg — the money left this asset.</summary>
    Out,

    /// <summary>The target's Deposit leg — the money arrived in this asset.</summary>
    In,
}

File: web/src/app/features/transactions/transfer-direction.ts
import type { TransactionTransferResponse, TransferDirection } from '../../api/portfolio';

// Backend enum Skarbiec.Portfolio.Features.Transfers.TransferDirection, serialized as its int in C#
// declaration order (asset-transfers-deposit-funding).
export const TRANSFER_DIRECTION = {
  Out: 0,
  In: 1,
} as const satisfies Record<string, TransferDirection>;

// The label a transfer leg carries beside its type: "Transfer to <asset> (<portfolio>)" on the Out
// leg, "Transfer from …" on the In leg.
export function transferLabel(transfer: TransactionTransferResponse): string {
  const direction = Number(transfer.direction) === TRANSFER_DIRECTION.Out ? 'to' : 'from';
  return `Transfer ${direction} ${transfer.counterpartAssetName} (${transfer.counterpartPortfolioName})`;
}
